File: Java/P002.java
// Project Euler
// Problem 2: The sum of Fibonacci number under 4000000

public class P002 {
    public static void main(String[] args) {
        int limit = 4000_000;

        System.out.println(compute(limit));
    }

    public static String compute(int limit) {
        int x = 0;  // the first  number of Fibonacci
        int y = 1;  // the second number of Fibonacci
        int temp = 0;   // temporary variable to store value
        int sum = 0;

        while (y < limit) {
            if (y % 2 == 0) sum += y;

            temp = x;
            x = y;
            y = x + temp;
        }

        return Integer.toString(sum);
    }
}

File: Active/TypeScript/src/P002/P002.ts
// Project Euler                                       
// Problem 2: The sum of even Fibonacci number under 4000000

export function compute002(limit: number): number {
    let sum: number = 0;

    let x: number = 1; // first digit of Fibonacci
    let y: number = 1; // second digit of Fibonacci
    let temp: number = 0; // temp variable to store value

    while (y < limit) {
        if (y % 2 == 0) {
            sum += y;
        }

        temp = x;       // store the first digit in temp
        x = y;          // store the second digit in x
        y = temp + x;   // derivate the third digit by adding temp and x
    }

    return sum;
}

function run002(): void {
    const limit: number = 4000000;
    console.log(compute002(limit).toString());
}

run002();
